applications/HDF5Application/custom_processes/hdf5_xdmf_connectivities_writer_process.cpp

```
@@ -29,13 +29,12 @@ XdmfConnectivitiesWriterProcess::XdmfConnectivitiesWriterProcess(const std::stri
     const int num_points = mpFile->GetDataDimensions(node_ids_path)[0];
     mpFile->ReadDataSet(node_ids_path, node_ids, 0, num_points);
 
+    mKratosToXdmfIdMap.reserve(num_points);
     // Set the parametric coordinate ids.
-    mKratosToXdmfIdTable.resize(num_points);
-#pragma omp parallel for
-    for (int i = 0; i < num_points; ++i)
-    {
-        // Here we expect the ids are one-based indices.
-        mKratosToXdmfIdTable(node_ids[i] - 1) = i;
+    for (int i = 0; i < num_points; ++i) {
+        const bool inserted = mKratosToXdmfIdMap.insert(IdMapType::value_type(node_ids[i], i)).second;
+        KRATOS_WARNING_IF("Kratos-Xdmf-Map", !inserted) << "Node-Id #" << node_ids[i]
+            << " is duplicated and only inserted once!" << std::endl;
```

We require a one-to-one mapping of ids onto indices in order to generate the xdmf connectivities. If this is not satisfied, we should throw an error instead of a warning:

```
const auto p = mKratosToXdmfIdMap.insert(IdMapType::value_type(node_ids[i], i));
KRATOS_ERROR_IF_NOT(p->second) << "Node #" << node_ids[i] << " at position " << i << " already exists at position " p->first->second << std::endl;
```

---

done, thx